float32 radius
float32 speed
float32 duration
float64 timestamp
